sbol_utilities/sbol3_sbol2_conversion.py

```diff
@@ -175,24 +175,36 @@ def visit_component(self, cp3: sbol3.Component):
                     sbol3.SBO_SIMPLE_CHEMICAL: sbol2.BIOPAX_SMALL_MOLECULE,
                     sbol3.SBO_NON_COVALENT_COMPLEX: sbol2.BIOPAX_COMPLEX}
         types2 = [type_map.get(t, t) for t in cp3.types]
-        # Make the Component object and add it to the document
-        cp2 = sbol2.ComponentDefinition(self._sbol2_identity(cp3), types2, version=self._sbol2_version(cp3))
-        self.doc2.addComponentDefinition(cp2)
-        # Convert the Component properties not covered by the constructor
-        cp2.roles = cp3.roles
-        cp2.sequences = cp3.sequences
-        if cp3.features:
-            raise NotImplementedError('Conversion of Component features from SBOL3 to SBOL2 not yet implemented')
-        if cp3.interactions:
-            raise NotImplementedError('Conversion of Component interactions from SBOL3 to SBOL2 not yet implemented')
-        if cp3.constraints:
-            raise NotImplementedError('Conversion of Component constraints from SBOL3 to SBOL2 not yet implemented')
-        if cp3.interface:
-            raise NotImplementedError('Conversion of Component interface from SBOL3 to SBOL2 not yet implemented')
-        if cp3.models:
-            raise NotImplementedError('Conversion of Component models from SBOL3 to SBOL2 not yet implemented')
-        # Map over all other TopLevel properties and extensions not covered by the constructor
-        self._convert_toplevel(cp3, cp2)
+
+        # Determine whether Component maps to a ComponentDefinition or ModuleDefinition
+        if sbol3.SBO_FUNCTIONAL_ENTITY not in cp3.types:
```

Comment:
We need to take more than the type into account in making this decision. An SBOL3 `Component` may actually map into SBOL2 as _both_ a ComponentDefinition and a ModuleDefinition. 

- If there are any `interactions`, `interface`, or `models`, or SBO function entity type, then there must by a `ModuleDefinition`
- If there are any `constraints`, `sequences`, or features with a `Location`, then there must be a `ComponentDefinition`.

Reply:
To keep this PR from ballooning and maintain issue scope to the linked issue (#293 ) which deals with converting a ModuleDefinition, I would suggest maybe inserting a TODO and a note in (#246 ) which concerns converting a Component.  Does that sound reasonable?

Reply:
I'm OK with inserting a TODO that explains this.